src/main/java/com/courtcanva/ccfranchise/services/FranchiseeService.java

```diff
@@ -0,0 +1,30 @@
+package com.courtcanva.ccfranchise.services;
+
+import com.courtcanva.ccfranchise.dtos.FranchiseeDto;
+import com.courtcanva.ccfranchise.model.Franchisee;
+import com.courtcanva.ccfranchise.repositories.FranchiseeRepository;
+import com.courtcanva.ccfranchise.utility.mappers.FranchiseeMapper;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class FranchiseeService {
+    private final FranchiseeRepository franchiseeRepository;
+
+    private final FranchiseeMapper franchiseeMapper;
+
+    public Long createFranchisee(FranchiseeDto franchiseeDto) {
+
+        Franchisee franchisee = franchiseeMapper.toFranchiseeEntity(franchiseeDto);
+
+        var newFranchisee = franchiseeRepository.save(franchisee);
+
+        return newFranchisee.getId();
+    }
+
+    public Franchisee findFranchiseeByABN(Integer abn){
+
+        return franchiseeRepository.findByABN(abn).orElseThrow(RuntimeException::new);
```

Comment:
We should define an explicit exception extending `RuntimeException` for this scenario, and also pass in the corresponding error message.

Reply:
Last morning I have introduced ControllerExceptionHandler and now it's a good time to practice it.

Reply:
create ErrorDto and ControllerExceptionHandler